### PR TITLE
More assertions

### PR DESCRIPTION
**Pull request by tomphp**

### CODE REVIEW

**Comment by localheinz** [comment]
Blocked by https://github.com/tomphp/config-service-provider/pull/37.

File: src/FileReader/PHPFileReader.php [hunk 1]
```diff
@@ -11,6 +12,8 @@
 
     public function read($filename)
     {
+        Assertion::string($filename);
+
         $this->filename = $filename;
 
         $this->assertFileExists();
```
Comment (localheinz) [inline src/FileReader/PHPFileReader.php]: @tomphp 

Was just thinking, should we use `Assertion::file()` here instead?

Comment (localheinz) [inline src/FileReader/PHPFileReader.php]: @tomphp 

And maybe even `Assertion::isArray()` for the `$config` we try to read from the file?

File: src/FileReader/ReaderFactory.php [hunk 2]
```diff
@@ -25,12 +27,16 @@ public function __construct(array $config)
...
-     * @param mixed $filename
+     * @param string $filename
      *
      * @return FileReader
+     *
+     * @throws InvalidArgumentException
      */
     public function create($filename)
     {
+        Assertion::string($filename);
```
Comment (localheinz) [inline src/FileReader/ReaderFactory.php]: @tomphp 

Should we also use `Assertion::file()`?

File: src/FileReader/PHPFileReader.php (hunk 1 above)
Comment (tomphp) [inline src/FileReader/PHPFileReader.php]: 👍  - will make it so

**Comment by localheinz** [comment]
@tomphp 

👍 Other than the suggestion to maybe use even more assertions (and throwing away some of our code and exceptions), this looks good to me!

File: src/FileReader/PHPFileReader.php (hunk 1 above)
Comment (tomphp) [inline src/FileReader/PHPFileReader.php]: @localheinz have asserted for `file` on `$filename` but not keen on the idea of `isArray` on `$config` as it throws an `InvalidArgumentException` and I'm not convince that's right for the file contents. Thoughts?

File: src/FileReader/FileReader.php [hunk 3]
```diff
@@ -8,6 +8,8 @@
      * @param string $filename
      *
      * @return array
+     *
+     * @throws InvalidArgumentException
```
Comment (localheinz) [inline src/FileReader/FileReader.php]: @tomphp 

This class is unknown here, shall we import it?

**Comment by localheinz** [comment]
@tomphp 

👍Other than https://github.com/tomphp/config-service-provider/pull/41/files#r78507110, this looks very good - feel free to merge once adjusted!

**Pull request merged**